include/node.hpp

```diff
@@ -229,12 +229,16 @@ class node : protected std::tuple<Arguments...> {
     }
 
 public:
-    constexpr node() noexcept : node(self()) {}
+    node() noexcept : node({}) {}
 
-    constexpr node(Derived &derived) noexcept
+    node(std::initializer_list<std::pair<const std::string, pmtv::pmt>> init_parameter) noexcept
         : _tags_at_input(traits::node::input_port_types<Derived>::size())
         , _tags_at_output(traits::node::output_port_types<Derived>::size())
-        , _settings(std::make_unique<basic_settings<Derived>>(derived)) {}
+        , _settings(std::make_unique<basic_settings<Derived>>(*static_cast<Derived *>(this))) {
```

Comment:
passing `this` which is a pointer to a non-constructed-yet object can lead to issues. I'd like to see a comment explaining why it is ok here, or something to be refactored so that this is not needed

Reply:
safe delegated use of this (i.e. not used during construction)

The `this` is needed when accessing/transferring the staged and active settings to/from the struct which is and can only be done after construction (unfortunately).